### ISSUE TITLE
UX improvement: add info that the cache contains images for other architectures than the current one

### ISSUE THREAD

**Issue by fatso83** [docker/for-mac#5899]
- [x] I have tried with the latest version of Docker Desktop
  - [ ] I have tried disabling enabled experimental features
  - [ ] I have uploaded Diagnostics
  - Diagnostics ID:

This bug is essentially piecing together various bug report (#5419, #5873) and [SO questions](https://stackoverflow.com/questions/68611904/unable-to-find-local-image-in-cache-when-building-derived-image) that all arise from Docker giving neither the right nor the relevant information to the user when not finding an image matching the current platform. Instead of going straight for the remote registry, it should see if there are any cached images matching the description for some other platform and posting a warning that 

> An image matching nicecorp-empty-db:latest was found in the cache for a different architecture. If you intend to use this, specify `--platform=linux/amd64` in your FROM definitions or set ` DOCKER_DEFAULT_PLATFORM=linux/amd64` as an environent variable".

### Expected behavior
- That the docker images build on an ARM (M1) based Mac as they would on an Intel based Mac
- That if a docker image cannot be found for the current architecture, Docker would inform you that a locally cached image exists that matches for another CPU platform

### Actual behavior
Bombs out with an error when trying to locate the referenced image upon which to build the new image
```
 => => transferring context: 2B                                                                                                                           0.0s
 => ERROR [internal] load metadata for docker.io/library/nicecorp-empty-db:latest                                                                           1.3s
------
 > [internal] load metadata for docker.io/library/nicecorp-empty-db:latest:
------
failed to solve with frontend dockerfile.v0: failed to create LLB definition: pull access denied, repository does not exist or may require authorization: server message: insufficient_scope: authorization failed
```

### Information

  - macOS Version: 11.5.2
  - Intel chip or Apple chip: M1
  - Docker Desktop Version: 3.5.2

### Steps to reproduce the behavior

  1. `docker build . -t nimble-empty-db`
  2. ` docker build -f Dockerfile-local-dev .`

You can also trigger the same issue on Intel platforms (here Ubuntu x86) by changing the `DOCKER_DEFAULT_PLATFORM`:

<details>

```
$ docker build -f Dockerfile-local-dev .
[+] Building 0.1s (8/8) FINISHED
 => [internal] load build definition from Dockerfile-local-dev                                                                   0.0s
 => => transferring dockerfile: 665B                                                                                             0.0s
 => [internal] load .dockerignore                                                                                                0.0s
 => => transferring context: 2B                                                                                                  0.0s
 => [internal] load metadata for docker.io/library/nicecorp-empty-db:latest                                                        0.0s
 => [1/4] FROM docker.io/library/nicecorp-empty-db                                                                                 0.0s
 => CACHED [2/4] RUN /etc/init.d/postgresql start &&     psql --command "CREATE DATABASE nicecorpdb_test_template OWNER nicecorpadm  0.0s
 => CACHED [3/4] RUN echo "host all all samenet password" >> /etc/postgresql/12/main/pg_hba.conf                                 0.0s
 => CACHED [4/4] RUN echo "max_connections = 1000" >> /etc/postgresql/12/main/postgresql.conf                                    0.0s
 => exporting to image                                                                                                           0.0s
 => => exporting layers                                                                                                          0.0s
 => => writing image sha256:43d315504f13aefa892d287ab14934f0fb5a1e610e66c947d6ca9ad54640c157                                     0.0s

Use 'docker scan' to run Snyk tests against images to find vulnerabilities and learn how to fix them

myuser at FAT-WORKSTATION in ~/dev/nicecorp/database (develop)
$ DOCKER_DEFAULT_PLATFORM=linux/arm64 docker build -f Dockerfile-local-dev .

[+] Building 2.4s (4/4) FINISHED
 => [internal] load build definition from Dockerfile-local-dev                                                                   0.0s
 => => transferring dockerfile: 48B                                                                                              0.0s
 => [internal] load .dockerignore                                                                                                0.0s
 => => transferring context: 2B                                                                                                  0.0s
 => ERROR [internal] load metadata for docker.io/library/nicecorp-empty-db:latest                                                  2.3s
 => [auth] library/nicecorp-empty-db:pull token for registry-1.docker.io                                                           0.0s
------
 > [internal] load metadata for docker.io/library/nicecorp-empty-db:latest:
------
failed to solve with frontend dockerfile.v0: failed to create LLB definition: pull access denied, repository does not exist or may require authorization: server message: insufficient_scope: authorization failed
```

</details>

# Docker files

<details><summary>Dockerfiles used</summary>

**Dockerfile**
```
FROM ubuntu:20.04

MAINTAINER ACME <devs@ACME.com>

RUN apt-get update && \
    apt-get install -y gnupg

# Manually set timezone so that apt-get install doesn't ask for it
ENV TZ=Europe/Oslo
RUN ln -snf /usr/share/zoneinfo/$TZ /etc/localtime && echo $TZ > /etc/timezone

### Postgres install and setup
RUN apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys B97B0AFCAA1A47F044F244A07FCC7D46ACCC4CF8 && \
    echo "deb http://apt.postgresql.org/pub/repos/apt/ focal-pgdg main" > /etc/apt/sources.list.d/pgdg.list && \
    apt-get update && \
    apt-get install -y postgresql-12 postgresql-client-12 postgresql-contrib-12 && \
    rm -rf /var/lib/apt/lists/*

USER postgres

RUN /etc/init.d/postgresql start && \
    # nicecorp Clinic setup
    psql --command "CREATE ROLE nicecorp LOGIN PASSWORD 'nicecorp';" && \
    psql --command "CREATE ROLE nicecorpadmin LOGIN PASSWORD 'nicecorpadmin';" &&\
    psql --command "CREATE DATABASE nicecorpdb OWNER nicecorpadmin;" &&\
    # nicecorp People setup
    psql --command "CREATE ROLE people LOGIN PASSWORD 'people';" &&\
    psql --command "CREATE ROLE peopleadmin LOGIN PASSWORD 'peopleadmin';" &&\
    psql --command "CREATE DATABASE peopledb OWNER peopleadmin;"

# Create custom pg_hba.conf
RUN echo "host nicecorpdb all samenet password" >> /etc/postgresql/12/main/pg_hba.conf && \
    echo "host peopledb all samenet password" >> /etc/postgresql/12/main/pg_hba.conf && \
    echo "listen_addresses='0.0.0.0'" >> /etc/postgresql/12/main/postgresql.conf

HEALTHCHECK --interval=1s --timeout=1s --start-period=1s CMD pg_isready

ENTRYPOINT ["bash"]
CMD ["-c", "/etc/init.d/postgresql start && tail -f /var/log/postgresql/postgresql-12-main.log"]
```

**Dockerfile-local-dev**
```
# Most definitely not to be used in production!
FROM nicecorp-empty-db

MAINTAINER ACME <devs@ACME.com>

# Allow nicecorpadmin user to create and drop databases (used for test runs)
RUN /etc/init.d/postgresql start && \
    psql --command "CREATE DATABASE nicecorpdb_test_template OWNER nicecorpadmin;" && \
    psql --command "ALTER USER nicecorpadmin SUPERUSER;"

# Allow all users to connect to all databases (used for test runs to allow ad-hoc databases)
RUN echo "host all all samenet password" >> /etc/postgresql/12/main/pg_hba.conf
RUN echo "max_connections = 1000" >> /etc/postgresql/12/main/postgresql.conf
```

</details>

**Comment by stephen-turner** [comment]
I'm not sure I've quite understood your proposal. I would think that the correct behaviour is to go to Hub rather than to use an image of the wrong architecture that you happen to have locally. Using images of the wrong architecture is best-effort only: they may crash, and they will always be less performant. We recommend they are used only as a last resort if the image can't be rebuilt for the correct architecture.

**Comment by fatso83** [comment]
>  I would think that the correct behaviour is to go to Hub rather than to use an image of the wrong architecture that you happen to have locally.

I did not mean to say automatically resorting to the closest fit is the solution, or even a solution. This issue is not about solving the fact that Docker could not find the image. The _actual problem_ is that most users will have no idea what goes wrong and Docker does nothing to aid them in finding the resolution, resorting to unexpected error messages that do not point the user in the right direction.

All I am saying is to say something to the user in the way of : "We can't find the image in the local cache, so we will try to find the image in the repository. By the way, we did find this image with exactly that name, but for another architecture. That coincidence does seem a bit too random, so check this documentation link for what to do and our best practices dealing with architectural issues" (not in those words).

This is akin to what most CLI tools do these days, trying to help the user find out what she really meant:
```
$ git statsu
git: 'statsu' is not a git command. See 'git --help'.

The most similar command is
        status
```
